CMakeLists.txt

```diff
@@ -192,6 +192,14 @@ cmake_dependent_option(
 
 set(ONNX_NAMESPACE "onnx_torch" CACHE STRING "A namespace for ONNX; needed to build with other frameworks that share ONNX.")
 
+# This is a fix for a rare build issue on Ubuntu:
+# symbol lookup error: miniconda3/envs/pytorch-py3.7/lib/libmkl_intel_lp64.so: undefined symbol: mkl_blas_dsyrk
+# https://software.intel.com/en-us/articles/symbol-lookup-error-when-linking-intel-mkl-with-gcc-on-ubuntu
+if(NOT APPLE)
+  set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "-Wl,--no-as-needed")
+  set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "-Wl,--no-as-needed")
```

Comment:
Adding `--no-as-needed` looks reasonable to me. My comments:

- One single line of `set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-as-needed")` would be effective on all build types.
- What would happen on Windows?

Reply:
> One single line of set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-as-needed") would be effective on all build types.
Addressed.

> What would happen on Windows?
I think it's a platform specific issue and platform specific fix(ubuntu/mkl/ld). Windows builds passed, but I restricted if condition to `if(UNIX AND NOT APPLE)`, see below

> Don't think it hurts to have this line on MacOS, and therefore no need of if(NOT APPLE).
It does: `ld: unknown option: --no-as-needed`